core/server/adapters/scheduling/post-scheduling/index.js

```diff
@@ -161,12 +161,17 @@ exports.init = function init(options = {}) {
                     adapter.schedule(_private.normalize({model, apiUrl, integration, resourceType: resource}));
                 });
 
+                /** We want to do reschedule as (unschedule + schedule) due to how token(+url) is generated
+                 * We want to first remove existing schedule by generating a matching token(+url)
+                 * followed by generating a new token(+url) for the new schedule
+                */
                 common.events.on(`${resource}.rescheduled`, (model) => {
-                    adapter.reschedule(_private.normalize({model, apiUrl, integration, resourceType: resource}));
+                    adapter.unschedule(_private.normalize({model, apiUrl, integration, resourceType: resource}, 'unscheduled'));
```

Comment:
@rishabhgrg  If I'm understanding this correctly, to remove the existing schedule we'd always need to pass an object formed using `'unscheduled'` parameter. Would this mean the same behavior is needed in https://github.com/TryGhost/Ghost/pull/11406/files#diff-4726ce3c4d18d41afad4b46cb0aa7dd3R151 ?

If the answer is yes, do we need to keep the `reschedule` method in the adapter at all? :thinking:

Reply:
> Would this mean the same behavior is needed in

@gargol Yes! Missed this one, good catch. 😄 

> If the answer is yes, do we need to keep the reschedule method in the adapter at all? 🤔

No there is no need to keep the reschedule method in adapter after this change, as since we changed how we generate URLs using token, calling just `reschedule` will never work properly as it just returns a single URL, which is not enough to delete existing job and add a new one, both with different URLs. Didn't remove it immediately from code, but 🤔 maybe we should just clean this up now. Also realized we should update the doc here as well - https://ghost.org/docs/concepts/custom-schedulers/